library/catalog/models.py

```diff
@@ -42,25 +43,35 @@ class MongoUser(Document):
     wishlist = ListField(default=[])
     preference = EmbeddedDocumentField(Preference.__name__, default=Preference())
 
-    def set_password(self, password):
-
-        self.django_password = make_password(password)
-        self.password_hash = generate_password_hash(password)
+    def generate_passwords(self, password):
+        return generate_password_hash(password), make_password(password)
 
     def check_password(self, password):
-        return check_password_hash(self.password_hash, password)
+        return check_password_hash(self.password, password)
 
     def save(self):
+        if self.password:
+            self.password, django_password = self.generate_passwords(self.password)
         mongo_user = super().save()
-        User = get_user_model()
-        django_user = User(username=self.username, email=self.email, password=self.django_password)
-        django_user.save()
+        get_user_model().objects.create_user(username=self.username,
+                                             email=self.email,
+                                             password=django_password)
         return mongo_user
 
     def update(self, **kwargs):
-        mongo_user = super().update(**kwargs)
-        User = get_user_model()
-        User.objects.filter(username=self.username).update(**kwargs)
+        mongo_kwargs = copy(kwargs)
```

Comment:
Why do we need to copy kwargs in this case ?

Reply:
We need to use a different set of arguments to update mongo and django models. That whay I copy the kwargs for each models and modify them